### PR TITLE
Migration guidelines

### PR DESCRIPTION
**Pull request by pepibumur**
In the aim of helping the users to adopt Tuist in their current projects, I'm adding some guidelines to the documentation.

### CODE REVIEW

File: website/markdown/docs/usage/adoption-guidelines.mdx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+# Adoption guidelines
+
+Unless you bootstrap a new project using Tuist,
+in which case you get everything configured automatically,
+you'll have to define your Xcode projects using Tuist's primitives _(e.g. Project.swift and Workspace.swift files)_.
+How tedious this process is,
+depends on how complex your projects are.
+
+As you probably know,
```
Comment (andreacipriani) [inline website/markdown/docs/usage/adoption-guidelines.mdx:14]: I'd rephrase this one, with something more mild, like: "as you probably know, Xcode projects can become very complex in time"

File: website/markdown/docs/usage/adoption-guidelines.mdx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+The leaner the projects, the easier they are to migrate.
+To make them thinner,
+you can extract information like build settings that can be defined outside Xcode projects.
+You must prevent developers from defining settings using build settings.
+You can do that by implementing a script that reads the build settings using [XcodeProj](https://github.com/tuist/xcodeproj),
+and fails if the settings of projects and targets are not empty.
+
+#### Migrate the most dependant projects first
+
+Those are usually simpler and contain fewer dependencies than the rest.
```
Comment (andreacipriani) [inline website/markdown/docs/usage/adoption-guidelines.mdx:50]: Do you want to mention `tuist graph` here as a command to use to see the dependency graph for a target? Also, we could easily build a script that outputs targets by dependencies

File: website/markdown/docs/usage/adoption-guidelines.mdx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+Although Xcode's build system ignores them, they might complicate the migration.
+
+#### Remove files that are not used by any target
+
+There might be files that were once part of the project,
+but that they are no longer needed.
+Find and remove them.
+Otherwise,
+you might run into compilation issues because your glob patterns _(e.g. `**/*.swift`)_ ended up matching them,
+and adding them as sources of your targets.
```
Comment (andreacipriani) [inline website/markdown/docs/usage/adoption-guidelines.mdx:65]: we can build a script for this that:
1) generates the project
2) uses xcdiff to compare the generated project against the base project and outputs the files that are in the generated one and **not** in the base one; and pipes them to `rm` :)

File: website/markdown/docs/usage/adoption-guidelines.mdx [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+#### Prevent modifications of Xcode projects after they've been migrated
+
+After the migration of each project,
+add a script that fails CI if someone modifies the Xcode project directly,
+and lets developers know that they should change the manifest files instead.
+
+You can use a tool like [xcdiff](https://github.com/bloomberg/xcdiff) for that.
```
Comment (andreacipriani) [inline website/markdown/docs/usage/adoption-guidelines.mdx:73]: if you are still checking in your pbxprojs on git, you can also go with a simple solution:
1) generate the project on CI
2) compare it with the project committed in the PR
if they are not the same file, fail the PR

File: website/markdown/docs/usage/adoption-guidelines.mdx (hunk 2 above)
Comment (pepibumur) [reply website/markdown/docs/usage/adoption-guidelines.mdx:50]: It'd be great to have a similar command that works with Xcode projects. I doubt `tuist graph` is useful until you have fully migrated.

**Comment by natanrolnik** [comment]
This is a very welcome addition to the docs, thanks @pepibumur

File: website/markdown/docs/usage/adoption-guidelines.mdx [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+You can use this snippet for that:
+
+```bash
+xcodebuild -scheme "MyScheme" -showBuildSettings >> MyScheme.xcconfig
+```
+
+#### Migrate the most dependant projects first
+
+Those are usually simpler and contain fewer dependencies than the rest.
+That makes them good candidates from which we can start the migration.
```
Comment (fortmarek) [inline website/markdown/docs/usage/adoption-guidelines.mdx:55]: I'd also mention that it is possible, if you have a more complicated project with multiple `.xcodeproj`s, to do the migration gradually. In the workspace you can just point to the path where `.xcodeproj` is generated and do it step-by-step.

**Comment by fortmarek** [comment]
Woops, I have not noticed that this has already been merged before I got to finish the review (I had dinner in the meantime) 🙈
